### PR TITLE
Add `append` and `format` option in `LogReport`

### PR DESCRIPTION
**Pull request by asi1024**
TODO: add tests

### CODE REVIEW

File: pytorch_pfn_extras/training/extensions/log_report.py [hunk 1]
```diff
@@ -12,8 +12,22 @@
...
+class LogWriterSaveFunc:
+
+    def __init__(self, format):
+        self._format = format
+
+    def __call__(self, target, file_o):
+        if self._format == 'json':
+            log = json.dumps(target, indent=4)
+        elif self._format == 'json-lines':
+            log = '\n'.join([json.dumps(x) for x in target])
```
Comment (kmaehashi) [inline pytorch_pfn_extras/training/extensions/log_report.py:24]: Could you support appending the result so that only diff is written when in append=True mode?

File: tests/pytorch_pfn_extras_tests/training_tests/extensions_tests/test_print_report.py [hunk 2]
```diff
@@ -3,18 +3,29 @@
...
     iters_per_epoch = 5
     manager = ppe.training.ExtensionsManager(
         {}, {}, max_epochs, iters_per_epoch=iters_per_epoch)
 
     out = io.StringIO()
-    log_report = ppe.training.extensions.LogReport()
+    log_report = extensions.LogReport(format=format, append=append)
     manager.extend(log_report)
-    extension = ppe.training.extensions.PrintReport(out=out)
+    extension = extensions.PrintReport(out=out)
```
Comment (emcastillo) [inline tests/pytorch_pfn_extras_tests/training_tests/extensions_tests/test_print_report.py:28]: I think this is not testing that the file is being written.
but it wasn't doing it before either. So we should do it in a different PR.

**Review by emcastillo — APPROVED** [review]

**Comment by emcastillo** [comment]
TODO: Add a proper `LogReport` test that writes the file and parses it back. 
There is currently no `LogReport` test, so I will be adding it in a different PR

**Comment by emcastillo** [comment]
CI fails are real :(
